lib/types/uuid.js

```diff
@@ -2,126 +2,150 @@
 
 const crypto = require("crypto");
 const utils = require("../utils");
+const rust = require("../../index");
 
 /** @module types */
 
 /**
- * Creates a new instance of Uuid based on a Buffer
  * @class
  * @classdesc Represents an immutable universally unique identifier (UUID). A UUID represents a 128-bit value.
- * @param {Buffer} buffer The 16-length buffer.
- * @constructor
  */
-function Uuid(buffer) {
-    if (!buffer || buffer.length !== 16) {
-        throw new Error("You must provide a buffer containing 16 bytes");
+class Uuid {
+    /**
+     * Creates a new instance of Uuid based on a Buffer
+     * @param {Buffer} buffer The 16-length buffer.
+     * @constructor
+     */
+    constructor(buffer) {
+        if (!buffer || buffer.length !== 16) {
+            throw new Error("You must provide a buffer containing 16 bytes");
+        }
+        this.internal = rust.UuidWrapper.new(buffer);
     }
-    this.buffer = buffer;
-}
 
-/**
- * Parses a string representation of a Uuid
- * @param {String} value
- * @returns {Uuid}
- */
-Uuid.fromString = function (value) {
-    //36 chars: 32 + 4 hyphens
-    if (typeof value !== "string" || value.length !== 36) {
-        throw new Error(
-            "Invalid string representation of Uuid, it should be in the 00000000-0000-0000-0000-000000000000",
+    /**
+     * Parses a string representation of a Uuid
+     * @param {String} value
+     * @returns {Uuid}
+     */
+    static fromString(value) {
+        // 36 chars: 32 + 4 hyphens
+        if (typeof value !== "string" || value.length !== 36) {
+            throw new Error(
+                "Invalid string representation of Uuid, it should be in the 00000000-0000-0000-0000-000000000000 format",
+            );
+        }
+        return new Uuid(
+            utils.allocBufferFromString(value.replace(/-/g, ""), "hex"),
```

Comment:
I know, this is how it worked in datastax, but maybe we want to to actually check it is in the correct format?
This should be a simple regex expression: https://stackoverflow.com/a/6640851